app/page.tsx

```diff
@@ -16,33 +16,86 @@ import ShowNutritionalTable from "@/components/ShowNutritionalTable";
 
 export default function Home() {
   const dispatch = useDispatch();
-
+  const currentRecipeId = "empty_recipe"; 
   const searchParams = useSearchParams();
   const params = useSelector((state: RootState) =>
-    selectURLParams(state, "empty_recipe")
+    selectURLParams(state,currentRecipeId)
   );
   React.useEffect(() => {
     dispatch(
       parseURLParameters({
-        recipeId: "empty_recipe",
+        recipeId: currentRecipeId,
         params: Array.from(searchParams.entries()).map(([key, value]) => ({
           key,
           value,
         })),
       })
     );
-  }, [dispatch, searchParams]);
+  }, [dispatch, searchParams, currentRecipeId]);
 
   const ingrdients = useSelector((state: RootState) =>
-    selectCurrentIngredients(state, "empty_recipe")
+    selectCurrentIngredients(state, currentRecipeId)
   );
-
+  function handleShareButtonClick() {
+    const url = "https://amathjourney.com/api/yololo";
+  
+    const params = selectURLParams(store.getState(), currentRecipeId);
+    const valuesAndQuantities = parseParams(params);
+    const body = JSON.stringify({ valuesAndQuantities });
+  
+    fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: body,
+    })
+      .then((response) => response.json())
+      .then((data) => {
+     return data
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }
+  function parseParams(params: string) {
+    const pairs = params.split("&");
+  
+    const valuesAndQuantities: { [x: string]: string | number; quantity: string; }[] = [];
+  
+    pairs.forEach((pair) => {
+      const [key, value] = pair.split("=");
+  
+      const [prefix, index] = key.split("_");
+      const quantityKey = `q${index}`;
+      const valueKey = `v${index}`;
+  
+      const existingObj = valuesAndQuantities.find((obj) => obj.quantity === value);
+      if (existingObj) {
+        existingObj[valueKey] = Number(value);
+      } else {
+        const newObj = {
+          quantity: value,
+          [quantityKey]: value,
+          [valueKey]: Number(value),
+        };
+        valuesAndQuantities.push(newObj);
+      }
+    });
+  
+    return valuesAndQuantities;
+  }
```

Comment:
Why do you bother to parse the URL params wherase you already have every thing well structured in the store?

If you look at `selectURLParams`, you will see it uses the `selectCurrentIngredients` and format the data to a URL params string.

Would be easier to do the same. Use `selectCurrentIngredients` and format the data to the object expected